code/game/mecha/equipment/tools/tools.dm

```diff
@@ -1309,5 +1332,85 @@
 		do_after_cooldown()
 	return 1
 
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool
+	name = "\improper Exosuit-Mounted Engineering Switchtool"
+	desc = "An exosuit-mounted Engineering switchtool. (Can be attached to: Any exosuit)"
+	icon_state = "mecha_rcd"	//needtochange
+	origin_tech = Tc_MATERIALS + "=3;" + Tc_POWERSTORAGE + "=2"
+	equip_cooldown = 10
+	energy_drain = 50
+	range = MELEE|RANGED
+	var/datum/global_iterator/pr_switchtool
+	var/obj/item/weapon/switchtool/engineering/mech/S
+
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool/New()
+	..()
+	S = new(src)
+	pr_switchtool = new /datum/global_iterator/mecha_switchtool(list(src),0)
+	pr_switchtool.set_delay(equip_cooldown)
+	pr_switchtool.toggle()
+
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool/Destroy()
+	qdel(S)
+	S = null
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool/action(atom/target)
+	if(S.deployed)
+		S.preattack(target, chassis.occupant, chassis.Adjacent(target))
+		chassis.use_power(energy_drain)
+
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool/Topic(href,href_list)
+	..()
+	if(href_list["change"])
+		if(S.deployed)
+			S.attack_self(chassis.occupant)
+		S.attack_self(chassis.occupant)
+	if(href_list["refill"])
+		pr_switchtool.toggle()
+		occupant_message("<span class='notice'>Automatic tool refilling activated.</span>")
+	update_equip_info()
+
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool/get_equip_info()
+	return "[..()] Current tool: [S.deployed ? "[S.deployed]" : "None"] \[<a href='?src=\ref[src];change=0'>change</a>\] [pr_switchtool.active() ? "" : "\[<a href='?src=\ref[src];refill=0'>activate refilling</a>\]"]"
+
+/datum/global_iterator/mecha_switchtool/process(var/obj/item/mecha_parts/mecha_equipment/tool/switchtool/ST)
+	if(!ST.chassis || ST.chassis.hasInternalDamage(MECHA_INT_SHORT_CIRCUIT))
+		if(ST.chassis)
+			ST.occupant_message("<span class='warning'>Electrical systems compromised. Automatic tool refilling deactivated.</span>")
+		stop()
+		ST.set_ready_state(1)
+		return
+	var/cur_charge = ST.chassis.get_charge()
+	if(isnull(cur_charge) || !ST.chassis.cell)
```

Comment:
The cell null check is redundant, `get_charge()` returns null if there is no cell already.